test/common/runtime/runtime_impl_test.cc

```diff
@@ -278,52 +284,189 @@ void testNewOverrides(Loader& loader, Stats::Store& store) {
-  loader->mergeValues({{"file2", ""}});
-  EXPECT_EQ("world", loader->snapshot().get("file2"));
-  EXPECT_EQ(0, store.gauge("runtime.admin_overrides_active").value());
+  loader_->mergeValues({{"file2", ""}});
+  EXPECT_EQ("world", loader_->snapshot().get("file2"));
+  EXPECT_EQ(0, store_.gauge("runtime.admin_overrides_active").value());
 
   // Override integer
-  loader->mergeValues({{"file3", "42"}});
-  EXPECT_EQ(42, loader->snapshot().getInteger("file3", 1));
-  EXPECT_EQ(1, store.gauge("runtime.admin_overrides_active").value());
+  loader_->mergeValues({{"file3", "42"}});
+  EXPECT_EQ(42, loader_->snapshot().getInteger("file3", 1));
+  EXPECT_EQ(1, store_.gauge("runtime.admin_overrides_active").value());
 
   // Remove overridden integer
-  loader->mergeValues({{"file3", ""}});
-  EXPECT_EQ(2, loader->snapshot().getInteger("file3", 1));
-  EXPECT_EQ(0, store.gauge("runtime.admin_overrides_active").value());
+  loader_->mergeValues({{"file3", ""}});
+  EXPECT_EQ(2, loader_->snapshot().getInteger("file3", 1));
+  EXPECT_EQ(0, store_.gauge("runtime.admin_overrides_active").value());
 
   // Override override string
-  loader->mergeValues({{"file1", "hello overridden override"}});
-  EXPECT_EQ("hello overridden override", loader->snapshot().get("file1"));
-  EXPECT_EQ(1, store.gauge("runtime.admin_overrides_active").value());
+  loader_->mergeValues({{"file1", "hello overridden override"}});
+  EXPECT_EQ("hello overridden override", loader_->snapshot().get("file1"));
+  EXPECT_EQ(1, store_.gauge("runtime.admin_overrides_active").value());
 
   // Remove overridden override string
-  loader->mergeValues({{"file1", ""}});
-  EXPECT_EQ("hello override", loader->snapshot().get("file1"));
-  EXPECT_EQ(0, store.gauge("runtime.admin_overrides_active").value());
+  loader_->mergeValues({{"file1", ""}});
+  EXPECT_EQ("hello override", loader_->snapshot().get("file1"));
+  EXPECT_EQ(0, store_.gauge("runtime.admin_overrides_active").value());
 }
 
-TEST(LoaderImplTest, All) {
-  MockRandomGenerator generator;
-  NiceMock<ThreadLocal::MockInstance> tls;
-  Stats::IsolatedStoreImpl store;
-  LoaderImpl loader(generator, store, tls);
-  EXPECT_EQ("", loader.snapshot().get("foo"));
-  EXPECT_EQ(1UL, loader.snapshot().getInteger("foo", 1));
-  EXPECT_CALL(generator, random()).WillOnce(Return(49));
-  EXPECT_TRUE(loader.snapshot().featureEnabled("foo", 50));
-  testNewOverrides(loader, store);
+// Validate that admin overrides disk, disk overrides bootstrap.
+TEST_F(DiskBackedLoaderImplTest, LayersOverride) {
+  base_ = TestUtility::parseYaml<ProtobufWkt::Struct>(R"EOF(
+    some: thing
+    other: thang
+    file2: whatevs
+  )EOF");
+  setup();
+  run("test/common/runtime/test_data/current", "envoy_override");
+  // Disk overrides bootstrap.
+  EXPECT_EQ("world", loader_->snapshot().get("file2"));
+  EXPECT_EQ("thing", loader_->snapshot().get("some"));
+  EXPECT_EQ("thang", loader_->snapshot().get("other"));
+  // Admin overrides disk and bootstrap.
```

Comment:
Is it admin, or latest takes?  If there's a file reload, does it overwrite the admin overrides?

I think we can be more clear about this in tests and docs.

Reply:
Good point. It overrides as a layer, I've added a test to demonstrate this (and convince myself!).